DatadogFlags/Sources/Client/FlagAssignmentsRequest.swift

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+extension URLRequest {
+    internal static func flagAssignmentsRequest(
+        url: URL,
+        evaluationContext: FlagsEvaluationContext,
+        context: DatadogContext,
+        customHeaders: [String: String]?
+    ) -> URLRequest {
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+
+        request.setValue("application/vnd.api+json", forHTTPHeaderField: "Content-Type")
+        request.setValue(context.clientToken, forHTTPHeaderField: "dd-client-token")
+
+        if let applicationId = context.additionalContext(ofType: RUMCoreContext.self)?.applicationID {
+            request.setValue(applicationId, forHTTPHeaderField: "dd-application-id")
+        }
+
+        if let customHeaders {
+            for (key, value) in customHeaders {
+                request.setValue(value, forHTTPHeaderField: key)
+            }
+        }
+
+        let requestBody = FlagAssignmentsRequestBody(
+            environment: FlagAssignmentsRequestBody.Environment(
+                name: context.env,
+                datadogEnvironment: context.env
+            ),
+            subject: FlagAssignmentsRequestBody.Subject(
+                targetingKey: evaluationContext.targetingKey,
+                targetingAttributes: evaluationContext.attributes
+            )
+        )
+
+        do {
+            let encoder = JSONEncoder.dd.default()
+            request.httpBody = try encoder.encode(requestBody)
+        } catch let error {
+            DD.logger.error("Failed to encode flag assignments request body.", error: error)
```

Comment:
suggestion/ It doesn’t seem useful for us to send a request without an httpBody in case of an encoding failure. We could improve this design by rethrowing the error and letting the caller handle the failure (immediate fail, no networking).

Reply:
Yes, I considered it at first, but ended up dropping that approach as `URLRequestBuilder` is focused on intake endpoints with compression. We can reconsider in the future making it more generic.